lib/LogicalEquivalence/Circuit.cpp

```diff
@@ -12,23 +12,166 @@
 //===----------------------------------------------------------------------===//
 
 #include "circt/LogicalEquivalence/Circuit.h"
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/Seq/SeqOps.h"
 #include "circt/LogicalEquivalence/LogicExporter.h"
 #include "circt/LogicalEquivalence/Solver.h"
 #include "circt/LogicalEquivalence/Utility.h"
 #include "mlir/IR/Builders.h"
 #include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/StringRef.h"
 
 #define DEBUG_TYPE "lec-circuit"
 
 using namespace mlir;
 using namespace circt;
 
+/// Populate the table of combinational transforms
+void Solver::Circuit::populateCombTransformTable() {
+  this->combTransformTable.insert(std::pair(
```

Comment:
It would also be nice to move this into another file, since a) it's a lot of code that, while vital, isn't _necessarily_ relevant to how the tool actually works algorithmically and b) a map that represents the semantics of different ops would be pretty nice to have available for other tools IMO

Reply:
@fabianschuiki Agreed, this is just a clunky workaround for not being able to walk directly, but fundamentally is the core of the LEC and MC logic - being able to encode it more elegantly would be great. Would also be lovely to have it reusable elsewhere - a dedicated collection of comb semantics seems pretty likely to end up being useful in other places.